### PR TITLE
refactor cli unit tests to specify app, credential, container, image, secret or volume to return in test (#868)

### PR DESCRIPTION
**Pull request by jsilverio22**
Per comments on PR #1046 this PR will allow devs to specify items to use in CLI unit tests. For more special cases, the switch statement can still be used however for general best practice defining the Item and adding to the MockClientFactory in your tests will allow for easier testing of CLI.

Issue: #868

### CODE REVIEW

File: pkg/cli/completion_test.go [hunk 1]
```diff
@@ -129,13 +129,29 @@ func TestAppsThenContainersCompletion(t *testing.T) {
 }
 
 func TestAppsCompletion(t *testing.T) {
+	var DefaultImageList = []apiv1.Image{{
+		TypeMeta:   metav1.TypeMeta{},
```
Comment (thedadams) [inline pkg/cli/completion_test.go:133]: nit: You don't need this like if `TypeMeta` is getting set to its "zero" value. Here and other tests altered in these changes.

File: pkg/cli/testdata/MockClient.go [hunk 2]
```diff
@@ -382,76 +451,24 @@ func (m *MockClient) VolumeDelete(ctx context.Context, name string) (*apiv1.Volu
 }
 
 func (m *MockClient) ImageList(ctx context.Context) ([]apiv1.Image, error) {
-	return []apiv1.Image{{
```
Comment (thedadams) [inline pkg/cli/testdata/MockClient.go:385]: Personally, I am a fan of leaving this in and just adding the `if` check at the beginning. That way, if a user only wants to use a basic list of images, they can do that with minimal effort.

You could also remove the `DefaultImagesList` from the tests in this case. What are your thoughts on this?

**Review by thedadams — APPROVED** [review]
A couple small comments. Overall I really like these changes. Thanks!

File: pkg/cli/testdata/MockClient.go (hunk 2 above)
Comment (jsilverio22) [reply pkg/cli/testdata/MockClient.go:385]: I was going back and forth on this so, I don't mind adding it back in

File: pkg/cli/completion_test.go (hunk 1 above)
Comment (jsilverio22) [reply pkg/cli/completion_test.go]: yeah, removed now

File: pkg/cli/testdata/MockClient.go [hunk 3]
```diff
@@ -400,58 +472,17 @@ func (m *MockClient) ImageList(ctx context.Context) ([]apiv1.Image, error) {
...
 func (m *MockClient) ImageDelete(ctx context.Context, name string, opts *client.ImageDeleteOptions) (*apiv1.Image, error) {
+	if m.ImageItem != nil {
+		return m.ImageItem, nil
+	}
 	switch name {
-	case "dne":
-		return nil, nil
-	case "found-image1234567":
-		return &apiv1.Image{}, nil
 	case "found-image-two-tags1234567":
```
Comment (tylerslaton) [inline pkg/cli/testdata/MockClient.go:486]: Maybe we could export `const` values for the common switch values that these can use? That would make the package a bit more discoverable for new devs coming in to write tests. That way as a developer I could name my object `testdata.FoundWithTwoTags` or something akin to that. Thoughts?

**Review by tylerslaton — APPROVED** [review]

**Pull request merged**